### PR TITLE
WT-3106 Add wt utility truncate operation

### PR DESCRIPTION
**Pull request by daveh86**
Not ready for review yet. Need to fix some comments and update documentation.

### CODE REVIEW

**Comment by daveh86** [comment]
@sulabhM can you have a look?

File: src/docs/command-line.dox [hunk 1]
```diff
@@ -370,6 +370,19 @@ Include only "fast" statistics in the output (equivalent to passing
 <code>statistics=(fast)</code>) to WT_SESSION::open_cursor.
 
 <hr>
+@section util_truncate wt truncate
+Trundate a table, removing all documents.
```
Comment (sulabhM) [inline src/docs/command-line.dox]: typo "Trundate"

File: test/suite/test_util14.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+import os
+from suite_subprocess import suite_subprocess
+import wiredtiger, wttest
+
+# test_util04.py
+#    Utilities: wt truncate
+class test_util04(wttest.WiredTigerTestCase, suite_subprocess):
+    tablename = 'test_util14.a'
+    nentries = 1000
```
Comment (sulabhM) [inline test/suite/test_util14.py]: this variable isn't used at all

File: test/suite/test_util14.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    def test_truncate_process(self):
+        """
+        Test truncate in a 'wt' process
+        """
+        params = 'key_format=S,value_format=S'
+        self.session.create('table:' + self.tablename, params)
+
```
Comment (sulabhM) [inline test/suite/test_util14.py]: Populate the table after create, maybe nentries is intended for that

File: test/suite/test_util14.py (hunk 2 above)
Comment (daveh86) [inline test/suite/test_util14.py]: Fixed

**Comment by daveh86** [comment]
Pushed a fix for the typo.

File: src/docs/command-line.dox [hunk 4]
```diff
@@ -370,6 +370,19 @@ Include only "fast" statistics in the output (equivalent to passing
 <code>statistics=(fast)</code>) to WT_SESSION::open_cursor.
 
 <hr>
+@section util_truncate wt truncate
+Truncate a table, removing all documents.
```
Comment (agorrod) [inline src/docs/command-line.dox]: ``documents`` is a mongodb-ism, I prefer ``data``.

File: test/suite/test_util14.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+from suite_subprocess import suite_subprocess
+import wiredtiger, wttest
+
+# test_util14.py
+#    Utilities: wt truncate
+class test_util14(wttest.WiredTigerTestCase, suite_subprocess):
+    tablename = 'test_util14.a'
+    nentries = 1000
+
+    def test_truncate_process(self):
```
Comment (agorrod) [inline test/suite/test_util14.py]: Thanks for creating a test case. I think it'd be worthwhile testing error cases here as well. Checking what happens if no table name is passed in, or a well-formed but invalid table name is passed in and other cases you can think of.

File: src/utilities/util_truncate.c [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+	argc -= __wt_optind;
+	argv += __wt_optind;
+
+	/* The remaining argument is the uri. */
+	if (argc != 1)
+		return (usage());
+	if ((name = util_name(session, *argv, "table")) == NULL)
+		return (1);
+
+	ret = session->truncate(session, name, NULL, NULL, NULL);
```
Comment (agorrod) [inline src/utilities/util_truncate.c]: I'd expect the command to print an error on failure here. For example ``util_create.c`` does:

```
        if ((ret = session->create(session, uri, config)) != 0)
                return (util_err(session, ret, "%s: session.create", uri));
```

Comment (daveh86) [inline src/utilities/util_truncate.c]: We also need to fix the util_drop code, as it follows the same pattern. I will raise a new issue shortly.

File: test/suite/test_util14.py (hunk 5 above)
Comment (daveh86) [inline test/suite/test_util14.py]: Added some fixes and new tests as suggested.

**Comment by agorrod** [comment]
I see an error when running s_all against this branch:

```
$ sh s_all
dist/s_all run started...
Updating files that include the package version
####################### MESSAGE ############################
s_all run of: "./s_python" resulted in:
    [tab] characters appear in Python scripts:
    	./test/suite/test_util14.py
#######################
dist/s_all run finished
```

I was surprised the pull request tester didn't detect the problem, so I re-ran the command used by the pull request tester, and don't see the error:

```
$ sh s_all -A -E
dist/s_all run started...
Updating files that include the package version
Rebuilding GNU tools library support
    configure: WARNING: DIAGNOSTIC BUILDS ARE NOT RECOMMENDED FOR PRODUCTION DEPLOYMENT.
    libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, `build_posix/gnu-support'.
    libtoolize: copying file `build_posix/gnu-support/ltmain.sh'
    configure.ac:23: installing 'build_posix/gnu-support/compile'
    configure.ac:31: installing 'build_posix/gnu-support/config.guess'
    configure.ac:31: installing 'build_posix/gnu-support/config.sub'
    configure.ac:16: installing 'build_posix/gnu-support/install-sh'
    configure.ac:16: installing 'build_posix/gnu-support/missing'
    Makefile.am: installing 'build_posix/gnu-support/depcomp'
    parallel-tests: installing 'build_posix/gnu-support/test-driver'
####################### MESSAGE ############################
s_all run of: "./s_export" resulted in:
    skipped: libwiredtiger.a not found
#######################
dist/s_all run finished
```

Could you open a ticket to get that problem addressed?

**Comment by agorrod** [comment]
Thanks @daveh86 lgtm.

**Pull request merged**